packages/ra-ui-materialui/src/button/ShowButton.tsx

```diff
@@ -45,13 +44,14 @@ ShowButton.propTypes = {
     record: PropTypes.any,
 };
 
-const enhance = shouldUpdate(
+const PureShowButton = memo(
+    ShowButton,
     (props: Props, nextProps: Props) =>
-        (props.record &&
-            nextProps.record &&
-            props.record.id !== nextProps.record.id) ||
-        props.basePath !== nextProps.basePath ||
-        (props.record == null && nextProps.record != null)
+        (!props.record ||
```

Comment:
This is also hard to read. The idea is that we want to compare record.id instead of the entire record (which may change although the values don't change), and the null check was added in 1ec8f2e895aa9c036b5c1031559223ac5ab59385 to avoid a bug. 

The right way to do it would be

```js
(props.record && nextProps.record ? props.record.id === nextProps.record.id : props.record == nextProps.record) &&
props.basePath === nextProps.basePath
```